concordium_p2p_rpc.proto

```diff
@@ -4,223 +4,438 @@ import "google/protobuf/wrappers.proto";
+  // This is *not* a transaction.
+  // A JSON schema for the return type is provided at: https://developer.concordium.software/en/mainnet/net/references/grpc.html.
   rpc InvokeContract(InvokeContractRequest) returns (JsonResponse) {}
+
+  // Get an overview of the balance of special accounts in the given block.
+  // A JSON schema for the return type is provided at: https://developer.concordium.software/en/mainnet/net/references/grpc.html.
   rpc GetRewardStatus(BlockHash) returns (JsonResponse) {}
+
+  // Get an overview of the parameters used for baking.
+  // A JSON schema for the return type is provided at: https://developer.concordium.software/en/mainnet/net/references/grpc.html.
   rpc GetBirkParameters(BlockHash) returns (JsonResponse) {}
+
+  // Get a list of smart contract modules that exist in the state after the given block.
+  // A JSON schema for the return type is provided at: https://developer.concordium.software/en/mainnet/net/references/grpc.html.
   rpc GetModuleList(BlockHash) returns (JsonResponse) {}
+
+  // Get the source of a smart contract module.
   rpc GetModuleSource(GetModuleSourceRequest) returns (BytesResponse) {}
+
+  // Get a list of all identity providers that exist in the state after the given block.
+  // A JSON schema for the return type is provided at: https://developer.concordium.software/en/mainnet/net/references/grpc.html.
   rpc GetIdentityProviders(BlockHash) returns (JsonResponse) {}
+
+  // Get a list of all anonymity revokers that exist in the state after the given block.
+  // A JSON schema for the return type is provided at: https://developer.concordium.software/en/mainnet/net/references/grpc.html.
   rpc GetAnonymityRevokers(BlockHash) returns (JsonResponse) {}
+
+  // Get the cryptographic parameters used in the given block.
+  // A JSON schema for the return type is provided at: https://developer.concordium.software/en/mainnet/net/references/grpc.html.
   rpc GetCryptographicParameters(BlockHash) returns (JsonResponse) {}
+
+  // Get a list of all baker IDs registered at that block in ascending order. Or null, if the block is invalid.
+  // A JSON schema for the return type is provided at: https://developer.concordium.software/en/mainnet/net/references/grpc.html.
   rpc GetBakerList(BlockHash) returns (JsonResponse) {}
+
+  // Get the status of a pool. If passiveDelegation == true, this returns the status for the passive delegators.
+  // Otherwise, it returns the status for the baker with the specified ID (if it exists).
+  // A JSON schema for the return type is provided at: https://developer.concordium.software/en/mainnet/net/references/grpc.html.
   rpc GetPoolStatus(GetPoolStatusRequest) returns (JsonResponse) {}
 
+  // Get a list of banned peers.
   rpc GetBannedPeers(Empty) returns (PeerListResponse) {}
+
+  // Shut down the node.
   rpc Shutdown(Empty) returns (BoolResponse) {}
+
+  // Start dumping packages into the specified file.
```

Comment:
This (and DumpStop) are only enabled if the node is built with the "network_dump" feature.